Feature get weather by location

MarsLviv opened the pull request.

MarsLviv committed:
enabling/disabling GetWeatherByLocation Button

MarsLviv committed:
class LocationClient: change getLocation signature - add callback in it; removed callback from constructor

MarsLviv committed:
class LocationClient: implement calback in getLocation

MarsLviv committed:
minor changes

gutachkee commented on app/src/main/java/com/example/weather/LocationClient.java:
```diff
@@ -23,36 +23,37 @@
     private final FusedLocationProviderClient fusedLocationClient;
     private final LocationRequest locationRequest;
-    private final LocationCallback locationCallback;
+    private /*final*/ LocationCallback locationCallback;
```
```suggestion
    private LocationCallback locationCallback;
```

gutachkee commented on app/src/main/java/com/example/weather/LocationClient.java:
```diff
@@ -23,36 +23,37 @@
+    private /*final*/ LocationCallback locationCallback;
 
-    private final RetrieveLocationCallback retrieveLocationCallback;
+    private /*final*/ RetrieveLocationCallback retrieveLocationCallback;
```
```suggestion
    private RetrieveLocationCallback retrieveLocationCallback;
```

gutachkee commented on app/src/main/java/com/example/weather/MainActivity.java:
```diff
@@ -114,9 +107,33 @@ public void onError() {
+        locationClient.getLocation(new LocationClient.RetrieveLocationCallback() {
+            @Override
+            public void onRetrieveLocation(@NonNull Location location) {
+                Log.d(TAG, "latitude:" + location.getLatitude() + "  longitude:" + location.getLongitude());
```
```suggestion
                Log.d(TAG, "onRetrieveLocation(): latitude:" + location.getLatitude() + "  longitude:" + location.getLongitude());
```

MarsLviv committed:
remove comments

gutachkee reviewed (approved):
few cosmetic change requests. Good job!

Consider to use lambdas where possible.

MarsLviv committed:
Refactoing:

-class WeatherService: -anonymous class replace to lambla;
-remove unused imports

MarsLviv committed:
Merge branch 'master' into feature-get_weather_by_location

MarsLviv committed:
Refactoring: class MainActivity: update log message
